### ISSUE TITLE
net::ERR_HTTP2_PROTOCOL_ERROR thrown when an update is available

### ISSUE THREAD

**Issue by CatalanCabbage** [electron-userland/electron-builder#4987]
* **Electron-Builder**: 22.6.0
* **Electron Version**: 7.1.11
* **Electron-Updater**: 4.3.1
* **Electron Type (current, beta, nightly)**: Current
* **Target**: Windows, nsis

I've set up electron-updater, and it works properly.
I trigger `checkForUpdate()` on app startup; however, I'm getting this error:

![image](https://user-images.githubusercontent.com/45961072/82331554-e83a4880-9a01-11ea-9857-fd9bdda5528c.png)

For crawlability, **Error**:
```
Uncaught Exception:
Error: net::ERR_HTTP2_PROTOCOL_ERROR
    at SimpleURLLoaderWrapper.<anonymous> (electron/js2c/browser_init.js:2545:21)
    at SimpleURLLoaderWrapper.emit (events.js:203:13)
```

**What have I noticed/tried?**
* Happens **only when a new update is available**. If no update is available, this error is not thrown.
* Flow: Open app > `checkForUpdates()`; update available > JavaScript error thrown > I press OK > Update happens properly, everything works as expected.
* No traces in DevTools: I open DevTools on Window creation, but no console errors/Network errors
* All searches for `ERR_HTTP2_PROTOCOL_ERROR` result in some Discord issue 😐. 
<br>However, I was also able to find threads that had various reasons for the error, over different apps, [running a merry dance around me](https://stackoverflow.com/questions/58215104/whats-the-neterr-http2-protocol-error-about) . 
* I tried setting extra headers such as `no-cache`, no luck. When I set `
autoUpdater.requestHeaders = { 'Transfer-encoding' : 'chunked'};`, the update didn't work, and the error didn't happen.

I'm opening the issue here since I'm guessing it's evidently an issue triggered by our request/download. If you'd like me to open it elsewhere, please do let me know.

**Comment by Venatum** [comment]
You can [disable http2](https://www.electronjs.org/docs/all#--disable-http2).

_main.js_
```js
app.commandLine.appendSwitch('disable-http2');
```

However another error like that may appear :

![image](https://user-images.githubusercontent.com/35234721/82898652-f801e180-9f59-11ea-9d5a-2b56f8d2da0f.png)

**Comment by CatalanCabbage** [comment]
Thanks, @Venatum, didn't know that existed!  
Also, I recall reading somewhere that `autoUpdater.requestHeaders = { 'Transfer-encoding' : 'chunked'};` prevents the Content Length Mismatch error; will try it out!

**Comment by CatalanCabbage** [comment]
I tried adding `app.commandLine.appendSwitch('disable-http2');` which resulted in another error:  

![image](https://user-images.githubusercontent.com/45961072/85210357-d29ba400-b35c-11ea-96f6-d7d4c506a1d4.png)

I read after some research again that disabling `cache` might fix this, so I tried [`--disable-http-cache`](https://www.electronjs.org/docs/all#--disable-http-cache) from the same `disable-http2` doc, which didn't work.

**Finally this worked:**
`autoUpdater.requestHeaders = {'Cache-Control' : 'no-store, no-cache, must-revalidate, proxy-revalidate, max-age=0'};`

### TLDR: 
*Set in `main.js`:*  
```
app.commandLine.appendSwitch('disable-http2');
autoUpdater.requestHeaders = {'Cache-Control' : 'no-store, no-cache, must-revalidate, proxy-revalidate, max-age=0'};
```
[Reference commit](https://gitlab.com/cataxcab/the-wall/-/commit/76ddbba028255798406f98f9a62bd74567333029)
